src/Symfony/Component/Cache/Traits/FilesystemCommonTrait.php

```diff
@@ -108,7 +108,7 @@ private function write(string $file, string $data, int $expiresAt = null)
             fwrite($h, $data);
             fclose($h);
 
-            if (null !== $expiresAt) {
+            if (null !== $expiresAt && 0 !== $expiresAt) {
```

Comment:
we should still touch() the file, but ahead in the future
let's say one year.
this to play well with
https://github.com/symfony/symfony/blob/3974dc112a9422ef3aea35bdd4cb432ddedd9d96/src/Symfony/Component/Cache/Traits/FilesystemTrait.php#L88

Reply:
So basically revert https://github.com/symfony/symfony/pull/26127/files#diff-4fe048bc7fe0fe862ad82f3e4eb15b3b2238d5ee696edd153cad6cd222bdf354L97 then? That would prob be cleaner than doing that within this method.

Reply:
nope, no revert, but instead of the current patch, change the next line to 
`touch($this->tmp, $expiresAt ?: time() + 31556952);`

Reply:
Perfect, that's also exactly what I came up with. It's pushed.